### PR TITLE
Adding attachments changes.

### PR DESCRIPTION
**Pull request by mkaundal-topleft**

### CODE REVIEW

**Comment by kti-cameron** [comment]
I'm getting Flake8 test fails when running `make test`. 

```
# flake8 config file at tox.ini
flake8 .
./djautotask/sync.py:1594:33: E231 missing whitespace after ','
./djautotask/sync.py:1595:80: E501 line too long (82 > 79 characters)
./djautotask/api.py:934:1: E302 expected 2 blank lines, found 1
./djautotask/api.py:940:5: E303 too many blank lines (2)
./djautotask/api.py:941:80: E501 line too long (85 > 79 characters)
./djautotask/api.py:966:28: E231 missing whitespace after ':'
./djautotask/api.py:966:35: E231 missing whitespace after ':'
./djautotask/api.py:966:40: E231 missing whitespace after ','
./djautotask/api.py:966:48: E231 missing whitespace after ':'
./djautotask/api.py:966:59: E231 missing whitespace after ','
./djautotask/api.py:966:80: E501 line too long (80 > 79 characters)
make: *** [Makefile:28: lint] Error 1
```
Can you fix those?

**Comment by mkaundal-topleft** [comment]
Sure.

**Comment by mkaundal-topleft** [comment]
Failing cases are fixed now.

File: djautotask/api.py [hunk 1]
```diff
@@ -930,3 +930,45 @@ def delete(self, parent, **kwargs):
...
+            if response.json().get('items'):
+                response = response.json().get('items')[0]
+
+            return response
+        else:
+            self._log_failed(response)
+            return None
+
+    def get_attachments(self, object_id, *args, **kwargs):
+        context = {"filter":[{"op":"eq","field":"ParentID","value": object_id}]}
```
Comment (kti-cameron) [inline djautotask/api.py]: Based on the details in https://www.autotask.net/help/developerhelp/Content/APIs/General/AttachmentChanges.htm it looks like using `ParentID` is no longer recommended and will be removed soon. So I think we'll need to replace `ParentID` with either `TaskID` or `TicketID`.

File: djautotask/api.py [hunk 2]
```diff
@@ -930,3 +930,45 @@ def delete(self, parent, **kwargs):
             kwargs.get("id")
         )
         return self.request('delete', endpoint_url)
+
+class SystemAPIClient(AutotaskAPIClient):
```
Comment (kti-cameron) [inline djautotask/api.py:935]: The name `SystemAPIClient` comes from CW so doesn't really make sense to use here. Instead lets call this `Attachment` or `AttachmentInfo` `APIClient`

File: djautotask/api.py (hunk 1 above)
Comment (mkaundal-topleft) [reply djautotask/api.py]: Updated the changes.

File: djautotask/api.py (hunk 2 above)
Comment (mkaundal-topleft) [reply djautotask/api.py]: Resolved.

File: djautotask/api.py [hunk 3]
```diff
@@ -930,3 +930,48 @@ def delete(self, parent, **kwargs):
...
+            if response.json().get('items'):
+                response = response.json().get('items')[0]
+
+            return response
+        else:
+            self._log_failed(response)
+            return None
+
+    def get_attachments(self, object_id, field, *args, **kwargs):
+        context = {"filter": [{"op": "eq",
```
Comment (kti-cameron) [inline djautotask/api.py:967]: I think there's a better way to structure the way these conditions are set for the request using some existing patterns. This does work fine for now. Just looking into it. I may make the change if it's simple.

File: djautotask/sync.py [hunk 4]
```diff
@@ -1575,6 +1579,42 @@ def _assign_field_data(self, instance, object_data):
         return instance
 
 
+class AttachmentSynchronizer:
+    client_class = api.AttachmentInfoAPIClient
+
+    def __init__(self, type, field=None, *args, **kwargs):
```
Comment (kti-cameron) [inline djautotask/sync.py:1585]: Should type be a keyword argument here too?

Comment (mkaundal-topleft) [reply djautotask/sync.py:1585]: Yes.

**Pull request merged**